astropy/coordinates/angles.py

```diff
@@ -396,9 +396,9 @@ def _wrap_at(self, wrap_angle):
         # any more for >= 1.19 (NUMPY_LT_1_19), but former is).
         with np.errstate(invalid='ignore'):
             wraps = (self_angle - wrap_angle_floor) // a360
-            np.nan_to_num(wraps, copy=False)
-            if np.any(wraps != 0):
-                self_angle -= wraps*a360
+            valid = np.isfinite(wraps) & (wraps != 0)
+            if np.any(valid):
+                self_angle[valid] -= wraps[valid] * a360
```

Comment:
Is this necessary? I'd have thought this would slow things down, at least for big arrays since now there will be loops over the indices inside numpy, and thought it wouldn't really affect possible `inf` and `nan`.

Reply:
There is a test case that creates a non writable array of nans that fails without the mask indexing

Reply:
Hmm, that is confusing. Why would that pass `np.any(valid)`?

(though, separately, perhaps we should just skip the whole attempt to wrap if the array is not writable?)

Reply:
See #12317 for the reason for this contortion.  I don't think you'd want to skip non-writeable arrays entirely, because you'd want the user to know why you aren't performing some needed wrapping.  But, you'd want to silently pass non-writeable arrays where no wrapping is needed.

Reply:
@ayshih - that makes sense but I am still confused how we can get here (i.e., some data points are finite and have non-zero wrap). 

But @maxnoe, I just ran the coordinates tests without the `[valid]` pieces and got no errors. Am I missing something?

Reply:
Ah, yes, the way `valid` is defined, there should be no need for the additional indexing to satisfy the test.

Reply:
Maybe I got the test failure on an intermediate version... week check

Reply:
I presume that if only a tiny fraction of the array elements (e.g., one out of a million) need to be wrapped, the indexing operation would perform better than adding zero to tons of elements.

Reply:
Ok, I checked and every case I tested (no nans, different proportions of nans, different numbers of elements) is faster without he indexing and the tests are passing.

Reply:
Yes, that makes sense. One still has to loop over all elements with the indexing, and the subtraction operation is cheap in comparison to looping over the index array twice.